js/woo.js

```diff
@@ -0,0 +1,1465 @@
+                        'code': networkId,
+                    };
+                }
+            }
+        }
+        return undefined;
+    }
+
+    currencyCodeWithNetwork (code, params, divider = '_', beforOrAfter = true) { // TODO: this method can be moved into base
+        // at first, try to get according to default networks
+        const defaultNetworkPair = this.getDefaultNetworkPairForCurrency (code, params);
+        if (defaultNetworkPair !== undefined) {
+            const id = defaultNetworkPair['id'];
+            let networkizedCode = undefined;
+            if (code === id) { // if there was no dash between pair, then its same
+                networkizedCode = code;
+            } else { // if there was dash, then needs to be joined
+                if (beforOrAfter) {
+                    networkizedCode = id + divider + code;
+                } else {
+                    networkizedCode = code + divider + id;
+                }
+            }
+            const networkCode = defaultNetworkPair['code'];
+            return [ networkizedCode, networkCode ];
+        } else {
+            // if it was not returned above according to defaults & options, then return the first network
+            const currencyItem = this.currencies[code];
+            const networks = currencyItem['networks'];
+            const networkKeys = Object.keys (networks);
+            for (let j = 0; j < networkKeys.length; j++) {
+                const networkCode = networkKeys[j];
+                const networkItem = networks[networkCode];
+                const networkizedCode = this.safeString (networkItem, 'chained_currency_code');
+                if (networkizedCode !== undefined) {
+                    return [ networkizedCode, networkCode ];
+                }
+            }
+        }
+        return undefined;
+    }
+
+    getCurrencyByNetworkizedCode (chainedCode) { // TODO: this method can be moved into base
+        const keys = Object.keys (this.currencies);
+        for (let i = 0; i < keys.length; i++) {
+            const key = keys[i];
+            const currencyItem = this.currencies[key];
+            const currencyCode = currencyItem['code'];
+            if (chainedCode === currencyCode) {
+                return chainedCode;
+            } else {
+                const networks = currencyItem['networks'];
+                const networkKeys = Object.keys (networks);
+                for (let j = 0; j < networkKeys.length; j++) {
+                    const networkKey = networkKeys[j];
+                    const networkItem = networks[networkKey];
+                    if (networkItem['chained_currency_code'] === chainedCode) {
+                        return currencyItem;
+                    }
+                }
+            }
+        }
+        return undefined;
+    }
+
+    objectMemberAt (objOrArray, position) { // TODO: can be in base
+        return objOrArray[Object.keys (objOrArray)[position]];
```

Comment:
Let's split all lines like this into multiple expressions, one expression per line. 

```JavaScript
const keys = Object.keys (objectOrArray);
const keyAtPosition = this.safeValue (keys, position);
return this.safeValue (objectOrArray, keyAtPosition);
```

In general, we can avoid this method entirely with the already existing basecode.